notebooks/forest_inference_demo.ipynb

```diff
@@ -43,129 +43,149 @@
+    "# synthetic data size\n",
+    "n_rows = 10000\n",
+    "n_columns = 100\n",
+    "n_categories = 2\n",
+    "random_state = cupy.random.RandomState(43210)\n",
     "\n",
-    "    # save the trained xgboost model\n",
-    "    bst.save_model(model_path)\n",
+    "# fraction of data used for model training\n",
+    "train_size = 0.8\n",
+    "\n",
+    "# trained model output filename\n",
+    "model_path = 'xgb.model'\n",
     "\n",
-    "    return bst"
+    "# num of iterations for which xgboost is trained\n",
+    "num_rounds = 100\n",
+    "\n",
+    "# maximum tree depth in each training round\n",
+    "max_depth = 20"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Predict helper function\n",
-    "Uses the trained xgboost model to perform prediction and return the labels."
+    "## Generate data"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 4,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def predict_xgboost_model(X_validation, y_validation, xgb_model):\n",
+    "# create the dataset\n",
+    "X, y = make_classification(\n",
+    "    n_samples=n_rows,\n",
+    "    n_features=n_columns,\n",
+    "    n_informative=int(n_columns/5),\n",
+    "    n_classes=n_categories,\n",
+    "    random_state=42\n",
+    ")\n",
     "\n",
-    "    # predict using the xgboost model\n",
-    "    dvalidation = xgb.DMatrix(X_validation, label=y_validation)\n",
-    "    xgb_preds = xgb_model.predict(dvalidation)\n",
+    "# convert the dataset to float32\n",
+    "X = X.astype('float32')\n",
+    "y = y.astype('float32')\n",
     "\n",
-    "    # convert the predicted values from xgboost into class labels\n",
-    "    xgb_preds = np.around(xgb_preds)\n",
-    "    return xgb_preds"
+    "# split the dataset into training and validation splits\n",
+    "X_train, X_validation, y_train, y_validation = train_test_split(X, y, train_size=0.8)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Define parameters"
+    "## Train helper function\n",
+    "Defines a simple function that trains the XGBoost model and returns the trained model.\n",
+    "\n",
+    "For additional information on the xgboost library please refer to the documentation on : \n",
+    "https://xgboost.readthedocs.io/en/latest/parameter.html"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 5,
    "metadata": {},
    "outputs": [],
    "source": [
-    "n_rows = 10000\n",
-    "n_columns = 100\n",
-    "n_categories = 2\n",
-    "random_state = np.random.RandomState(43210)\n",
+    "def train_xgboost_model(\n",
+    "    X_train, \n",
+    "    y_train,\n",
+    "    model_path='xgb.model',\n",
+    "    num_rounds=150, \n",
+    "    max_depth=20\n",
+    "):\n",
+    "    \n",
+    "    # set the xgboost model parameters\n",
+    "    params = {\n",
+    "        'silent': 0, \n",
```

Comment:
Is silent still a viable parameter in current xgboost? I didn't think so, but I could be mistaken.

Reply:
I will double check, thanks.

Reply:
The `silent` parameter has been deprecated. It is recommended to set `verbosity=0` or `verbosity=1` instead.

Reply:
It is now verbosity -- https://xgboost.readthedocs.io/en/latest/parameter.html#general-parameters